pkg/kv/kvserver/batcheval/cmd_end_transaction.go

```diff
@@ -1054,13 +1048,24 @@ func splitTrigger(
 			"unable to fetch original range mvcc stats for split")
 	}
 
+	// Only calculate the rangeKeyDelta if we are likely to need it.
+	rangeKeyDeltaRequired := currentStats.ContainsEstimates == 0 && bothDeltaMS.ContainsEstimates == 0
+	var rangeKeyDeltaMS enginepb.MVCCStats
+	if rangeKeyDeltaRequired {
+		rangeKeyDeltaMS, err = computeSplitRangeKeyStatsDelta(ctx, batch, split.LeftDesc, split.RightDesc)
+		if err != nil {
+			return enginepb.MVCCStats{}, result.Result{}, errors.Wrap(err,
+				"unable to compute range key stats delta for RHS")
+		}
+	}
```

Comment:
@msbutler You had a slightly different condition here, but reading the code, I think the condition above is the only time we care about getting the rangeKey stats. But please take a second look.